spec/beaker-pe/install/pe_utils_spec.rb

```diff
@@ -860,10 +860,11 @@ def slice_installer_options(host)
 
     it 'sets puppet-agent acceptable_exit_codes correctly for config helper on upgrade' do
       hosts = make_hosts({
-                           :previous_pe_ver => '3.0',
-                           :pe_ver => '4.0',
-                           :roles => ['agent'],
-                         }, 2)
+        :previous_pe_ver => '3.0',
```

Comment:
@kevpl ok, I've stared at this for a bit.  During upgrade, pe_ver gets replaced by pe_upgrade_ver here:

https://github.com/puppetlabs/beaker-pe/blob/1.6.1/lib/beaker-pe/install/pe_utils.rb#L836-L858

It then calls do_install(), just like a regular install run does.  Because do_install expects to be able to find the pe_ver on the host for what it is about to install, pe_ver is overwritten in this way.

This is all very messy, and ideally we'd refactor all of this and instead explicitly pass in what version we are installing, and leave some sane markers in the config alone (pe_ver being the original install, pe_upgrade_ver being the upgrade ver, or something).

But for how things stand now, setting pe_upgrade_ver in the spec is reasonable.

Other option is to tweak https://github.com/puppetlabs/beaker-pe/blob/1.6.1/lib/beaker-pe/install/pe_utils.rb#L124 to use pe_ver; I think I used pe_upgrade_ver for clarity ( :) ).  But since that is working and this is temporary code to be revisted in the next few weeks as we continue meep work for flanders, I'm inclined to just fix the spec as you have done.